### PR TITLE
WIP: Add support for tabindex, aria label, aria role, and aria hidden

### PR DESCRIPTION
**Pull request by domoritz**
**vega-parser**

- Add support for `ariaLabel`, `ariaRole`, `ariaHidden`, and `tabindex` to encodings, axes, legends, and titles.

**vega-schema**

- Add support for `ariaLabel`, `ariaRole`, `ariaHidden`, and `tabindex` to encodings, axes, legends, and titles.

**vega-scenegraph**

- Add support for `ariaLabel`, `ariaRole`, `ariaHidden`, and `tabindex` to SVG renderer.

**vega-typings**

- Add `ariaLabel`, `ariaRole`, `ariaHidden`, and `tabindex` to encodings, axes, legends, and titles. 

With these properties, we can make charts accessible (#268).

Here is an example spec. Note that these new properties only work with the SVG renderer. 

Needed for https://github.com/vega/vega-lite/issues/4244.

### CODE REVIEW

**Comment by domoritz** [comment]
We should also add aria-hidden as discussed in https://github.com/vega/vega-lite/issues/4244.

**Comment by domoritz** [comment]
A more extensive tests spec for aria hidden, aria group, and aria label.

```json
{
  "$schema": "https://vega.github.io/schema/vega/v5.json",
  "width": 400,
  "height": 200,
  "padding": 5,
  "description": "Bar chart showing amounts per category.",
  "title": {
    "text": "Interesting chart title",
    "ariaLabel": "the chart title",
    "subtitle": "An even more interesting subtitle",
    "subtitleAriaLabel": "the chart subtitle"
  },
  "data": [
    {
      "name": "table",
      "values": [
        {"category": "A", "amount": 28},
        {"category": "B", "amount": 55},
        {"category": "C", "amount": 43},
        {"category": "D", "amount": 91},
        {"category": "E", "amount": 81},
        {"category": "F", "amount": 53},
        {"category": "G", "amount": 19},
        {"category": "H", "amount": 87}
      ]
    }
  ],
  "scales": [
    {
      "name": "xscale",
      "type": "band",
      "domain": {"data": "table", "field": "category"},
      "range": "width",
      "padding": 0.05,
      "round": true
    },
    {
      "name": "yscale",
      "domain": {"data": "table", "field": "amount"},
      "nice": true,
      "range": "height"
    },
    {
      "name": "qcolor",
      "domain": {"data": "table", "field": "amount"},
      "range": "heatmap"
    }
  ],
  "axes": [
    {
      "orient": "bottom",
      "scale": "xscale",
      "title": "Category",
      "titleAriaLabel": "X axis title",
      "labelAriaLabel": "x label"
    },
    {
      "title": "Amount",
      "orient": "left",
      "scale": "yscale",
      "ariaHidden": true,
      "titleAriaLabel": "Y axis title",
      "labelAriaHidden": true
    }
  ],
  "legends": [{"fill": "qcolor"}],
  "marks": [
    {
      "type": "group",
      "role": "figure",
      "encode": {
        "enter": {
          "ariaLabel": {"value": "The bars of the chart"}
        }
      },
      "marks": [
        {
          "type": "rect",
          "from": {"data": "table"},
          "encode": {
            "enter": {
              "x": {"scale": "xscale", "field": "category"},
              "width": {"scale": "xscale", "band": 1},
              "y": {"scale": "yscale", "field": "amount"},
              "y2": {"scale": "yscale", "value": 0},
              "fill": {"scale": "qcolor", "field": "amount"},
              "ariaLabel": {"signal": "datum.category + ' ' + datum.amount"}
            }
          }
        }
      ]
    }
  ]
}
```

File: packages/vega-parser/src/parsers/axis.js [hunk 1]
```diff
@@ -44,6 +44,10 @@ export default function(spec, scope) {
       titlePadding: encoder(_('titlePadding')),
       minExtent:    encoder(_('minExtent')),
       maxExtent:    encoder(_('maxExtent')),
+      ariaLabel:    encoder(_('ariaLabel')),
+      ariaRole:     encoder(_('ariaRole')),
+      ariaHidden:   encoder(_('ariaHidden')),
+      tabindex:     encoder(_('tabindex')),
```
Comment (donghaoren) [inline packages/vega-parser/src/parsers/axis.js]: `tabIndex`?

File: packages/vega-parser/src/parsers/guides/axis-labels.js [hunk 2]
```diff
@@ -88,7 +88,11 @@ export default function(spec, config, userEncode, dataRef, size, band) {
     fontSize:    _('labelFontSize'),
     fontWeight:  _('labelFontWeight'),
     fontStyle:   _('labelFontStyle'),
-    limit:       _('labelLimit')
+    limit:       _('labelLimit'),
+    ariaLabel:   _('labelAriaLabel'),
+    ariaRole:    _('labelAriaRole'),
+    ariaHidden:  _('labelAriaHidden'),
+    tabindex:    _('labeltabindex')
```
Comment (donghaoren) [inline packages/vega-parser/src/parsers/guides/axis-labels.js]: `labelTabIndex`?

File: packages/vega-parser/src/parsers/axis.js (hunk 1 above)
Comment (domoritz) [reply packages/vega-parser/src/parsers/axis.js]: Maybe. we also have `zindex` and there is already `tabindex` on the web https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex.

**Comment by domoritz** [comment]
https://github.com/vega/vega/pull/2380 is better now

**Pull request closed without merging**